File: october circuits/countOfInt.cpp
#include <bits/stdc++.h>
using namespace std;

const long N = 100000001;

char *sieve;


void get_primes(unsigned long max){
    sieve = new char[max/8+1];
    memset(sieve, 0xFF, (max/8+1) * sizeof(char));
    for(unsigned long x = 2; x <= max; x++)
        if(sieve[x/8] & (0x01 << (x % 8)))
            for(unsigned long j = 2*x; j <= max; j += x)
                sieve[j/8] &= ~(0x01 << (j % 8));
}


int main() {
    get_primes(N);
    int tc, n, a;
    cin >> tc;
    while(tc--){
        cin >> n;
        int count = 0;
        while(n--){
            cin >> a;
            if(sieve[a/8] & (0x01 << (a % 8)))
                count++;
        }
        cout << count << '\n';
    }
    cout << flush;
    
}

File: october circuits/countOfInt.py
MAX = 10**6 + 1
prime = [True for i in range(MAX)] 
p = 2
while (p * p <= MAX): 
        
    if (prime[p] == True): 
        for i in range(p * 2, MAX, p): 
            prime[i] = False
    p += 1


for tc in range(int(input())):
    input()
    count = 0
    for val in [int(x) for x in input().split()]:
        if prime[val]: count += 1
    print(count)
